src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs

```diff
@@ -54,8 +61,62 @@ public object Convert(object value, Type targetType, object parameter, CultureIn
 
             var iconBitmapImage = new BitmapImage();
             iconBitmapImage.BeginInit();
-            iconBitmapImage.UriSource = iconUrl;
 
+            BitmapSource imageResult;
+
+            if (values.Length > 1
+                && values[1] != null
+                && values[1] != System.Windows.DependencyProperty.UnsetValue
+                && iconUrl.IsFile
+                && !string.IsNullOrEmpty(iconUrl.Fragment)
```

Comment:
Would IsNullOrWhitespace avoid a string of spaces being picked up as an iconUrl?
ie, what happens if I put "    " as my iconUrl?

Reply:
Excellent questions.

> Would IsNullOrWhitespace avoid a string of spaces being picked up as an iconUrl?

No. See answer to the next question.

> ie, what happens if I put " " as my iconUrl?

The Uri parsing will not allow you to create Uri objects with only spaces. For example: `var x = new Uri("   ")` will throw an exception.

The other case is that you specify your embedded icon with just spaces, i.e., in your nuspec, you have `<icon>    </icon>`. Here, `LocalPackageSearchMetadata.GetIcon()` will create an URI in the form `file://path/to/package#    `. This is parse by the `Uri` class and then, that is not a valid NuGet Embedded Icon Uri

Reply:
I modified the code to reflect these new findings